### PR TITLE
Add optional `ReadableStream` polyfill

### LINKED ISSUE #6: Polyfill ReadableStream
Would it make sense to polyfill readable stream in the node stream library or https://www.npmjs.com/package/web-streams-polyfill? Otherwise the onDownloadProgress option isn't usable for non-browsers

### PR DESCRIPTION
**Pull request by Richienb**
Closes #6 

cc @sindresorhus

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -62,7 +62,11 @@
 		"xo": "^0.25.3"
 	},
 	"peerDependencies": {
-		"ky": ">=0.17.0"
+		"ky": ">=0.17.0",
+		"web-streams-polyfill": ">=2.0.0"
+	},
+	"optionalDependencies": {
+		"web-streams-polyfill": ">=2.0.0"
```
Comment (sholladay) [inline package.json]: Why is it in both `optionalDependencies` and `peerDependencies`?

Doesn't this still print a warning at install time if the user's project omits `web-streams-polyfill` from their package.json? If so, I don't think there's any value in including it in `optionalDependencies`.

Comment (Richienb) [reply package.json]: In that case, I'll remove the peerDependency entry since it's not required to be installed.

File: index.js [hunk 2]
```diff
@@ -22,4 +31,8 @@ if (!global.AbortController) {
 	global.AbortController = AbortController;
 }
 
+if (!global.ReadableStream && streams) {
+	global.ReadableStream = streams.ReadableStream;
```
Comment (sindresorhus) [inline index.js]:
```suggestion
	try {
		global.ReadableStream = require('web-streams-polyfill/ponyfill/es2018');
	} catch (_) {}
```

File: package.json (hunk 1 above)
Comment (sindresorhus) [reply package.json]: `optionalDependencies` is for things that can fail to install, for example, native addons. `web-streams-polyfill` should not be a dependency. It should be an **optional peer dependency**

https://github.com/yarnpkg/rfcs/blob/master/accepted/0000-optional-peer-dependencies.md

File: readme.md [hunk 3]
```diff
@@ -23,6 +23,8 @@ $ npm install ky ky-universal
 
 *Note that you also need to install `ky`.*
 
+For `ReadableStream` support, also install [`web-streams-polyfill`](https://github.com/MattiasBuelens/web-streams-polyfill).
```
Comment (sindresorhus) [inline readme.md]: This should elaborate on how adding `ReadableStream` benefits Ky usage.

File: package.json (hunk 1 above)
Comment (sholladay) [reply package.json]: > It should be an optional peer dependency

I agree. It's interesting that you can configure the precise behavior of peer dependencies with Yarn, but should we use that feature? I don't use Yarn, so I don't have a strong opinion on that.

I think the decision is between forcing one of two things on ~npm~ most users:
 - Peer dependency -> logs a warning unless polyfill is listed in package.json
 - Optional dependency -> larger `node_modules` unless polyfill fails to install

Comment (Richienb) [reply package.json]: `peerDependenciesMeta` is actually supported on all node package managers now (npm, yarn and pnpm) so I've swapped out the metadata.

**Comment by Richienb** [comment]
@sindresorhus Please re-review.

File: readme.md [hunk 4]
```diff
@@ -23,6 +23,22 @@ $ npm install ky ky-universal
...
+const ky = require("ky-universal")
+
+(async () => {
+	const {body} = ky('https://httpbin.org/bytes/16');
+	const {value} = await body.getReader().read();
+	const result = new TextDecoder("utf-8").decode(value)
+	
+	// …
+})();
+```
```
Comment (sindresorhus) [inline readme.md]: I don't think this example should be above the main example.

File: readme.md [hunk 5]
```diff
@@ -23,6 +23,22 @@ $ npm install ky ky-universal
 
 *Note that you also need to install `ky`.*
 
+For `ReadableStream` support, also install [`web-streams-polyfill`](https://github.com/MattiasBuelens/web-streams-polyfill).
+
+You can then use it normally:
+
+```js
+const ky = require("ky-universal")
```
Comment (sindresorhus) [inline readme.md]:
```suggestion
const ky = require('ky-universal');
```

File: readme.md [hunk 6]
```diff
@@ -23,6 +23,22 @@ $ npm install ky ky-universal
...
+
+(async () => {
+	const {body} = ky('https://httpbin.org/bytes/16');
+	const {value} = await body.getReader().read();
+	const result = new TextDecoder("utf-8").decode(value)
```
Comment (sindresorhus) [inline readme.md]:
```suggestion
	const result = new TextDecoder('utf-8').decode(value);
```

File: readme.md (hunk 4 above)
Comment (Richienb) [reply readme.md]: Fixed!

File: readme.md [hunk 7]
```diff
@@ -35,6 +35,24 @@ const ky = require('ky-universal');
...
+```js
+const ky = require('ky-universal');
+
+(async () => {
+	const {body} = ky('https://httpbin.org/bytes/16');
+	const {value} = await body.getReader().read();
+	const result = new TextDecoder('utf-8').decode(value);
+	
+	// …
+})();
```
Comment (sindresorhus) [inline readme.md]: Can you run this example locally and ensure it works? I'm seeing at least one issue with a missing `await`.

Comment (Richienb) [reply readme.md]: Nice catch! The missing `await` seems to be the only problem.

**Comment by sindresorhus** [comment]
👍

**Pull request merged**